lib/scsocket.js

```diff
@@ -677,47 +677,51 @@ SCSocket.prototype._handleEventAckTimeout = function (eventObject, eventNode) {
   }
 };
 
-SCSocket.prototype._emit = function (event, data, callback) {
+SCSocket.prototype._emit = function (event, data) {
   var self = this;
 
-  if (this.state == this.CLOSED) {
-    this.connect();
-  }
-  var eventObject = {
-    event: event,
-    callback: callback
-  };
+  return new Promise(function(resolve,reject) {
+    if (self.state == self.CLOSED) {
+      self.connect();
+    }
+    var eventObject = {
+      event: event,
+      callback: function(e,d) {
+        if (e) return reject(e)
+        return resolve(d)
+      }
+    };
 
-  var eventNode = new LinkedList.Item();
+    var eventNode = new LinkedList.Item();
 
-  if (this.options.cloneData) {
-    eventObject.data = clone(data);
-  } else {
-    eventObject.data = data;
-  }
-  eventNode.data = eventObject;
+    if (self.options.cloneData) {
+      eventObject.data = clone(data);
+    } else {
+      eventObject.data = data;
+    }
+    eventNode.data = eventObject;
 
-  eventObject.timeout = setTimeout(function () {
-    self._handleEventAckTimeout(eventObject, eventNode);
-  }, this.ackTimeout);
+    eventObject.timeout = setTimeout(function () {
+      self._handleEventAckTimeout(eventObject, eventNode);
+    }, self.ackTimeout);
 
-  this._emitBuffer.append(eventNode);
+    self._emitBuffer.append(eventNode);
 
-  if (this.state == this.OPEN) {
-    this._flushEmitBuffer();
-  }
+    if (self.state == self.OPEN) {
+      self._flushEmitBuffer();
+    }
+  })
 };
 
 SCSocket.prototype.send = function (data) {
   this.transport.send(data);
 };
 
 SCSocket.prototype.emit = function (event, data, callback) {
-  if (this._localEvents[event] == null) {
-    this._emit(event, data, callback);
-  } else {
-    Emitter.prototype.emit.call(this, event, data);
-  }
+  if (this._localEvents[event] != null) return Emitter.prototype.emit.call(this, event, data)
```

Comment:
For consistency, we should always add brackets `{}` around `if` blocks (even if only a single line); this makes it easier to read and edit the code in the future. Also there should be a semicolon at the end of each line.